tests/chainerx_tests/unit_tests/test_array.py

```diff
@@ -706,6 +706,22 @@ def test_array_pickle(device):
         chainerx.array([1, 2], chainerx.float32))
 
 
+# TODO(niboshi): Add pickle test involving context destruction and re-creation
+@pytest.mark.parametrize_device(['native:0', 'native:1', 'cuda:0'])
+def test_array_pickle_device_name(device):
+    arr = chainerx.array([1, 2], chainerx.float32, device=device.name)
+    s = pickle.dumps(arr)
```

Comment:
This test succeeds in master.
The `device` instance must not exist in order for this test to fail.
`del device` would not work either, because pytest (or a pytest fixture) holds the device instance alive.

Reply:
One way I can think of is to extend the `device` fixture (defined in `chainerx_tests/conftest.py`) to have an option to give a device name instead of a device instance.

Reply:
Or perhaps it's better to make a new fixture (e.g. `device_name`).

Reply:
I think that just passing the device name as a parameter should be fine,
But the test will fail if there are no cuda devices. (This happened in travis)

Reply:
The third of my comments means `parametrize_device_name` would check the existence of devices, just as what `parametrize_device` does.

Reply:
Understood 😃